corehq/apps/locations/resources/v0_1.py

```diff
@@ -95,12 +65,15 @@ def obj_get_list(self, bundle, **kwargs):
                 raise BadRequest(LOCATION_ACCESS_DENIED)
             parent = get_location_or_not_exist(parent_id, domain)
             locs = self.child_queryset(domain, include_inactive, parent)
-        return [child for child in locs if child.location_id in viewable]
+        return [child for child in locs if user_can_view_location(user, child, project)]
```

Comment:
I think this should just skip this check entirely, unless `location_restriction_for_users == True`.